udata/features/territories/__init__.py

```diff
@@ -13,13 +13,13 @@ def check_for_territory(query):
         return False
     # If it's a code, try INSEE/postal, otherwise use the name.
     if len(query) == 5 and query.isdigit():
-        # First check INSEE code then postal code.
+        # First check INSEE code then the first postal code.
         try:
             return GeoZone.objects.get(code=query, level='fr/town')
         except GeoZone.DoesNotExist:
             try:
-                return GeoZone.objects.get(
-                    keys__postal__contains=query, level='fr/town')
+                return GeoZone.objects.filter(
```

Comment:
The `filter` is optionnal (I know you like the Django way ;))

To avoid long lines, you can factorize:
```python
qs = GeoZone.objects(level=fr/town)
try:
    return qs.get(code=query)
except GeoZone.DoesNotExist:
    try:
        return qs(keys__postal__contains=query).first()
    except:
        #...
```